### ISSUE TITLE
VGG with Variational Inference not training

### ISSUE THREAD

**Issue by kumar-shridhar** [xuanqing94/BayesianDefense#2]
Hi,
I am training a VGG VI network on CIFAR-10 and the validation accuracy remains very low (20%) even after training for 200 epochs. The model was overfitting with training accuracy reaching past 70. I put L2 regularization (weight_decay) in the optimizer but still no increase in validation accuracy.
Is there a reason for it? What am I doing wrong here? I used all the default parameters. 

Thanks,
Kumar

**Comment by kumar-shridhar** [comment]
Also, training the adversarial VI code gives the same results as opposed to what has been proposed in the paper.
![image](https://user-images.githubusercontent.com/20888097/54910919-95dcc500-4eed-11e9-8c29-e514c9a45910.png)

How to replicate the results? Is there something that needs to be done pre-hand?

**Comment by xuanqing94** [comment]
Just want to do a quick check: can you download the checkpoints provided in the Google drive link?

**Comment by kumar-shridhar** [comment]
Ok. I will do it now. I trained it from scratch.

**Comment by kumar-shridhar** [comment]
Hi. From the checkpoint (cifar_vgg_vi), I got 91% on CIFAR10 test data. Why is it that I cannot train from scratch and reach the same accuracy? How was the checkpoint created? Were there some hyper-parameters settings done?

**Comment by xuanqing94** [comment]
The hyper-parameters are all listed in the appendix, how do you run the script? because I didn't list the bash command to run cifar_vgg_vi in README, it is possible that parameters are not the same.

You might also try to run the script multiple times and pick a good one, although in my experience I don't find it necessary.

**Comment by kumar-shridhar** [comment]
I used the following hyper-parameters:
lr=0.01
sigma_0=0.15
init_s=0.15

**Comment by xuanqing94** [comment]
@kumar-shridhar Then we are using different setting, please check the parameters in appendix B.

**Comment by kumar-shridhar** [comment]
Hi @xuanqing94 ,
I used the checkpoints directly (cifar10_vgg_adv_vi.pth) and ran a fgsm attack (the simplest one) and seems like the model behaves same as a frequentist one. Here are my results:
![image](https://user-images.githubusercontent.com/20888097/55700770-8c824b00-5a0b-11e9-9c17-6edc926403af.png)

**Comment by xuanqing94** [comment]
@kumar-shridhar Seems that you didn't do aggregation. See ./acc_under_attack.py.
Also, under high distortions (>0.1), no model does well (~10%). I suggest following the settings in the paper and code.

**Comment by shamoons** [comment]
Where do you put the CIFAR-10 data? And which one did you download? The one for Python?
